sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_search_sessions_v1/query.sql

```diff
@@ -0,0 +1,215 @@
+WITH events_unnested AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    sample_id,
+    client_info.client_id,
+    timestamp AS event_timestamp,
+    normalized_channel,
+    normalized_country_code,
+    name AS event_name,
+    CASE
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN 'engaged'
+      WHEN name = 'abandonment'
+        THEN 'abandoned'
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") NOT IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN "annoyance"
+      ELSE NULL
+    END AS search_session_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',')[OFFSET(0)] AS result_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',') AS results,
+    mozfun.map.get_key(extra, "selected_result") AS selected_results,
+    mozfun.map.get_key(extra, "engagement_type") AS engagement_type,
+    mozfun.map.get_key(extra, "n_chars") AS number_of_chars_typed,
+    mozfun.map.get_key(extra, "n_results") AS num_total_results,
+    metrics,
+    metrics.uuid.legacy_telemetry_client_id AS legacy_telemetry_client_id,
+    ping_info.experiments
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.events_v1`,
+    UNNEST(events)
+  CROSS JOIN
+    UNNEST(ping_info.experiments) AS e
```

Comment:
I think that if you simply delete these 2 lines, then you won't have to ARRAY_CONCAT_AGG later. @alekhyamoz

Reply:
I tried that but was unable to group on ARRAY (experiments) later in the SQL.

Reply:
Ah, okay. Can we still delete these 2 lines, and then keep the ARRAY_CONCAT_AGG later? I think the result will be the same, and saves 2 lines of code / confusion that we might be duplicating rows.

Reply:
Another option is we could join experiments in after the final group by on (client_id, document_id).

Reply:
Looks like we do need to remove this unnest from here - if we unnest and then ARRAY_CONCAT_AGG later, all the countifs for impression counts are operating on the unnested/duplicated data. I'm seeing double digit impression counts for many of the result types.